### PR TITLE
[new] Announce command

### PR DESCRIPTION
**Pull request by sourTaste000**
i hate null safety

### CODE REVIEW

File: src/main/kotlin/commands/AnnounceCommand.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+object AnnounceCommand : Command("announce") {
+    init {
+        greedyString("content"){
+            doesLater{context ->
+                val content: String? = context arg "content"
+                if (content.isNullOrEmpty()) {
+                    message.error("You need to put announcement here.")
+                    return@doesLater
+                }
+                Main.client?.servers?.find(message.server?.id ?: run {
```
Comment (l1ving) [inline src/main/kotlin/commands/AnnounceCommand.kt:20]: yet **again** you're looking for the server for no reason at all. Replace this with the word server

File: src/main/kotlin/commands/AnnounceCommand.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+                    message.error("Run this command inside a server!")
```
Comment (l1ving) [inline src/main/kotlin/commands/AnnounceCommand.kt:21]: this literally can't happen

File: src/main/kotlin/commands/AnnounceCommand.kt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+                    return@doesLater
+                })?.textChannels?.find(readConfigSafe<UserConfig>(ConfigType.USER, false)?.announceChannel ?: run {
```
Comment (l1ving) [inline src/main/kotlin/commands/AnnounceCommand.kt:23]: you're missing a permission check

**Review by l1ving — CHANGES_REQUESTED** [review]

File: src/main/kotlin/commands/AnnounceCommand.kt (hunk 2 above)
Comment (humboldt123) [reply src/main/kotlin/commands/AnnounceCommand.kt:21]: It can happen.

Comment (l1ving) [reply src/main/kotlin/commands/AnnounceCommand.kt:21]: No, it cannot, because server cannot be null. The bot doesn't listen to messages in dms

Comment (humboldt123) [reply src/main/kotlin/commands/AnnounceCommand.kt:21]: Server can be null if ran in a DM

Comment (l1ving) [reply src/main/kotlin/commands/AnnounceCommand.kt:21]: It cannot be run in a dm.

File: src/main/kotlin/commands/AnnounceCommand.kt [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+object AnnounceCommand : Command("announce") {
+    init {
+        greedyString("content"){
+            doesLater{context ->
+                val content: String? = context arg "content"
+                if (content.isNullOrEmpty()) {
+                    message.error("You need to put announcement here.")
+                    return@doesLater
+                } else if (!message.hasPermission(PermissionTypes.COUNCIL_MEMBER)){
```
Comment (l1ving) [inline src/main/kotlin/commands/AnnounceCommand.kt:20]: can we make a new permission called ANNOUNCE or something? I'd like it separate

File: src/main/kotlin/commands/AnnounceCommand.kt [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+                    message.error("You don't have the permission to do that!")
+                    return@doesLater
+                }
+                message.server?.textChannels?.find(readConfigSafe<UserConfig>(ConfigType.USER, false)?.announceChannel ?: run {
+                    message.error("Bad announcement channel found in config!")
```
Comment (l1ving) [inline src/main/kotlin/commands/AnnounceCommand.kt:25]: change this to ```"Please configure the `announcementChannel` in the `${ConfigType.USER.configPath.substring(7)}` config!"```

File: src/main/kotlin/commands/AnnounceCommand.kt [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+                    return@doesLater
+                })?.send(content)
```
Comment (l1ving) [inline src/main/kotlin/commands/AnnounceCommand.kt:27]: why not send an embed with the color set to Colors.primary?

Comment (sourTaste000) [reply src/main/kotlin/commands/AnnounceCommand.kt:27]: what should the title be

File: src/main/kotlin/commands/AnnounceCommand.kt [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+import arg
+import doesLater
+
+object AnnounceCommand : Command("announce") {
+    init {
+        greedyString("content"){
+            doesLater{context ->
+                val content: String = context arg "content"
+                if (!message.hasPermission(PermissionTypes.ANNOUNCE)){
+                    message.error("You don't have the permission to do that!")
```
Comment (l1ving) [inline src/main/kotlin/commands/AnnounceCommand.kt:19]: remove this, hasPermission already sends feedback

File: src/main/kotlin/commands/AnnounceCommand.kt [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+                }
+                message.server?.textChannels?.find(readConfigSafe<UserConfig>(ConfigType.USER, false)?.announceChannel ?: run {
+                    message.error("Please configure the `announcementChannel` in the `${ConfigType.USER.configPath.substring(7)}` config!")
+                    return@doesLater
+                })?.send {
+                    embed {
+                        description = content
+                        color = Colors.primary
+                    }
+                }
```
Comment (l1ving) [inline src/main/kotlin/commands/AnnounceCommand.kt:30]: `?: run { message.error("Error sending message! Channel couldn't be found") }`

**Review by l1ving — CHANGES_REQUESTED** [review]
Add error checking if the bot isn't allowed to talk in that channel please

**Review by l1ving — APPROVED** [review]
lgtm, I'll autoformat and merge this later

**Pull request merged**